omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,220 @@
+
+import org.openmrs.api.APIException;
+import org.openmrs.module.webservices.helper.TaskAction;
+import org.openmrs.module.webservices.helper.TaskServiceWrapper;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Creatable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+
+import javax.servlet.ServletContext;
+import java.util.ArrayList;
+import java.util.Collection;
+
+@Resource(name = RestConstants.VERSION_1 + "/taskaction", supportedClass = TaskAction.class, supportedOpenmrsVersions = {
+        "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class TaskActionResource1_8 extends BaseDelegatingResource<TaskAction> implements Creatable {
+	
+	private TaskServiceWrapper taskServiceWrapper = new TaskServiceWrapper();
+	
+	public void setTaskServiceWrapper(TaskServiceWrapper taskServiceWrapper) {
+		this.taskServiceWrapper = taskServiceWrapper;
+	}
+	
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		TaskAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<TaskDefinition> taskDefinitions;
+		if (action.isAllTasks() != null && action.isAllTasks()) {
+			taskDefinitions = taskServiceWrapper.getRegisteredTasks();
+			action.setTasks(new ArrayList<TaskDefinition>(taskDefinitions));
+		} else {
+			taskDefinitions = action.getTasks();
+		}
+		ServletContext servletContext = getServletContext(context);
+		
+		if (action.getAction() == TaskAction.Action.SCHEDULETASK || action.getAction() == TaskAction.Action.RESCHEDULETASK
+		        || action.getAction() == TaskAction.Action.DELETE || action.getAction() == TaskAction.Action.SHUTDOWNTASK) {
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException("Cannot execute action " + action.getAction()
+				        + " on empty set of task definitions.");
+			}
+		}
+		if (action.isAllTasks() == null || !action.isAllTasks()) {
```

Comment:
Like i said above, i do not find it API friendly for a boolean function to require checking for null. It should either be true or false.

Reply:
I will default assign isAllTasks = false

Reply:
Change its type too.